src/schema/notifications.ts

```diff
@@ -49,6 +67,91 @@ export class ZimbraNotifications {
 		this.handleFolderNotifications(notification);
 		this.handleConversationNotifications(notification);
 		this.handleMessageNotifications(notification);
+		this.handleContactNotifications(notification);
+	};
+
+	private handleContactNotifications = (notification: Notification) => {
+		const items = itemsForKey(notification, 'cn');
+		if (items) {
+			let searchResponse: any = {};
+			items.forEach((i: any) => {
+				const item = normalizeContact(i);
+				const defaultFolderName = 'Contacts';
+				const folder = this.cache.readFragment({
+					id: `Folder:${item.folderId}`,
+					fragment: gql`
+					fragment folderName${item.folderId} on Folder {
+						name
+					}
+					`
+				});
+				const folderName = (folder && folder.name) || defaultFolderName;
+				const group =
+					folderName === 'Trash'
+						? ''
+						: item.attributes.type === 'group'
+							? ' #type:group'
+							: ' NOT #type:group';
+				const query = `in:\\\\"${folderName}\\\\"${group}`;
+				const r = new RegExp(query);
+				if (!searchResponse[query]) {
+					searchResponse[query] = this.cache.readFragment({
+						id: `${findDataId(this.cache, '$ROOT_QUERY.search', dataId =>
```

Comment:
The template string is redundant since the string has only one part.

```js
id: findDataId(this.cache, '$ROOT_QUERY.search', dataId => r.test(dataId))
```